packages/curl.rb

```diff
@@ -3,108 +3,237 @@
+        CXXFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans#{@arch_ssp_cflags}' \
+        CPPFLAGS='-I#{@deppath}/include' \
+        LDFLAGS='-L#{@deppath}/lib'"
+
+    FileUtils.mkdir_p 'build/zlib'
+    Dir.chdir 'build/zlib' do
+      puts 'Building Zlib.'.lightgreen
+      system 'curl -Ls http://zlib.net/zlib-1.2.11.tar.gz | tar --strip-components=1 -zxf -'
+      system "#{@curldep_env_options} ./configure --prefix=#{@deppath} \
+        --static"
+      system 'make'
+      system 'make install'
+    end
+    FileUtils.mkdir_p 'build/zstd'
+    Dir.chdir 'build/zstd' do
+      puts 'Building Zstd.'.lightgreen
+      system 'curl -Ls https://github.com/facebook/zstd/archive/v1.5.0.tar.gz | tar --strip-components=1 -zxf -'
+      FileUtils.mkdir('build/cmake/builddir')
+      Dir.chdir('build/cmake/builddir') do
+        system "#{@curldep_cmake_options} \
+        -DZSTD_BUILD_STATIC=ON \
+        -DZSTD_BUILD_SHARED=OFF \
+        ../ -G Ninja"
+      end
+      system 'ninja -C build/cmake/builddir'
+      system 'ninja -C build/cmake/builddir install'
+    end
+    FileUtils.mkdir_p 'build/brotli'
+    Dir.chdir 'build/brotli' do
+      puts 'Building Brotli.'.lightgreen
+      system 'curl -Ls https://github.com/google/brotli/archive/v1.0.9.tar.gz | tar --strip-components=1 -zxf -'
+      FileUtils.mkdir('builddir')
+      Dir.chdir('builddir') do
+        system "#{@curldep_cmake_options} \
+          -DBUILD_SHARED_LIBS=OFF \
+          -DWITH_STATIC_LIB=ON \
+        ../ -G Ninja"
+      end
+      system 'ninja -C builddir'
+      system 'ninja -C builddir install'
+      Dir.chdir "#{@deppath}/lib" do
+        @brotlilibs = %w[libbrotlidec libbrotlienc libbrotlicommon]
+        @brotlilibs.each do |lib|
+          FileUtils.ln_s "#{lib}-static.a", "#{lib}.a"
+        end
+      end
+    end
+    FileUtils.mkdir_p 'build/idn2'
+    Dir.chdir 'build/idn2' do
+      puts 'Building IDN2.'.lightgreen
+      system 'curl -Ls http://ftp.gnu.org/gnu/libidn/libidn2-2.3.1.tar.gz | tar --strip-components=1 -zxf -'
+      system "#{@curldep_env_options} ./configure --prefix=#{@deppath} \
+        --disable-shared"
+      system 'make'
+      system 'make install'
+    end
+    FileUtils.mkdir_p 'build/ssl'
+    case ARCH
+    when 'aarch64', 'armv7l'
+      @arch_c_flags = '-march=armv7-a -mfloat-abi=hard'
+      @arch_cxx_flags = '-march=armv7-a -mfloat-abi=hard'
+      @openssl_configure_target = 'linux-generic32'
+    when 'i686'
+      @arch_c_flags = ''
+      @arch_cxx_flags = ''
+      @openssl_configure_target = 'linux-x86'
+    when 'x86_64'
+      @arch_c_flags = ''
+      @arch_cxx_flags = ''
+      @openssl_configure_target = 'linux-x86_64'
+    end
+    Dir.chdir 'build/ssl' do
+      puts 'Building OpenSSL.'.lightgreen
+      system 'curl -Ls https://www.openssl.org/source/openssl-1.1.1k.tar.gz | tar --strip-components=1 -zxf -'
```

Comment:
Why do we need to build openssl?  Why not just reference the package instead?

Reply:
We need to link to openssl static libs for https to work. The segfault I was getting was from accessing https URLs.

Building all of these dependencies separately, and with musl, ensured no dependence upon glibc.

It's really hard to debug curl issues unless you go dependency by dependency, and doing everything with musl not only took glibc out of the equation, but made debugging this issue a hell of a lot easier.

Curl is by no means the longest build of any package anyways, even with these dependency builds.